src/test/compile-fail/on-unimplemented/slice-index.rs

```diff
@@ -17,12 +17,12 @@ use std::ops::Index;
 #[rustc_error]
 fn main() {
     let x = &[1, 2, 3] as &[i32];
-    x[1i32];
-    //~^ ERROR E0277
-    //~| NOTE the trait `std::ops::Index<i32>` is not implemented for `[i32]`
-    //~| NOTE slice indices are of type `usize`
-    x[..1i32];
-    //~^ ERROR E0277
-    //~| NOTE the trait `std::ops::Index<std::ops::RangeTo<i32>>` is not implemented for `[i32]`
-    //~| NOTE slice indices are of type `usize`
+    x[1i32]; //~ ERROR E0277
+             //~| NOTE slice indices are of type `usize` or ranges of `usize`
+             //~| NOTE trait `std::slice::SliceIndex<i32>` is not implemented for `i32`
+             //~| NOTE required because of the requirements on the impl of `std::ops::Index<i32>`
+    x[..1i32]; //~ ERROR E0277
+               //~| NOTE slice indices are of type `usize` or ranges of `usize`
+               //~| NOTE trait `std::slice::SliceIndex<i32>` is not implemented for `std::ops::RangeTo<i32>`
```

Comment:
tidy failed, `line longer than 100 chars` (should not impact the crater run though).